Source/FieldSolver/MagnetostaticSolver/MagnetostaticSolver.cpp

```diff
@@ -65,7 +65,11 @@ WarpX::ComputeMagnetostaticField()
     // Fields have been reset in Electrostatic solver for this time step, these fields
     // are added into the B fields after electrostatic solve
 
-    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(this->max_level == 0, "Magnetostatic solver not implemented with mesh refinement.");
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(this->max_level == 0,
+        "Magnetostatic solver not implemented with mesh refinement.");
+
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(EB::enabled(),
+        "Magnetostatic Solver currently requires an embedded boundary.");
```

Comment:
`EB::enabled()` only returns `true` if an embedded boundary is actually installed, right? But this fix really only needs EB support to be on when compiling WarpX (which will eventually be enforced). In the meantime though I think this issue would be better fixed by doing:
```suggestion
#ifndef AMREX_USE_EB
    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(false,
        "Magnetostatic Solver currently requires an embedded boundary.");
#endif
```

I know we are trying to move away from using the compiler directives but I think this special circumstance justifies it.

Reply:
I will double check if this impacts 3D cases as well, but I had to actually install an EB and set the boundary outside of the domain to keep it from crashing in AMReX. So I would need EB::enabled() to be true. I guess the message needs to be more verbose though to explain it better.

Reply:
I have updated the assert to reflect that the solver works when EB support is not compiled, but when compiled with EB there are functions in AMReX that have not been updated to work when EB support is compiled but not installed.

Please see AMReX Issue https://github.com/AMReX-Codes/amrex/issues/4223